python/ctsm/site_and_regional/single_point_case.py

```diff
@@ -0,0 +1,429 @@
+        logger.info("fluse_in:  %s", fluse_in)
+        logger.info("fluse_out: %s", os.path.join(self.out_dir, fluse_out))
+
+        # create 1d coordinate variables to enable sel() method
+        f_in = self.create_1d_coord(
+            fluse_in, "LONGXY", "LATIXY", "lsmlon", "lsmlat"
+        )
+
+        # extract gridcell closest to plon/plat
+        f_out = f_in.sel(lsmlon=self.plon, lsmlat=self.plat, method="nearest")
+
+        # expand dimensions
+        f_out = f_out.expand_dims(["lsmlat", "lsmlon"])
+
+        # specify dimension order
+        f_out = f_out.transpose(u"time", u"cft", u"natpft", u"lsmlat", u"lsmlon")
+
+        # revert expand dimensions of YEAR
+        year = np.squeeze(np.asarray(f_out["YEAR"]))
+        temp_xr = xr.DataArray(year, coords={"time": f_out["time"]}, dims="time", name="YEAR")
+        temp_xr.attrs["units"] = "unitless"
+        temp_xr.attrs["long_name"] = "Year of PFT data"
+        f_out["YEAR"] = temp_xr
+
+        # update attributes
+        self.update_metadata(f_out)
+        f_out.attrs["Created_from"] = fluse_in
+
+        wfile = os.path.join(self.out_dir, fluse_out)
+        # mode 'w' overwrites file
+        f_out.to_netcdf(path=wfile, mode="w", format="NETCDF3_64BIT")
+        logger.info("Successfully created file (fluse_out), %s", wfile)
+        f_in.close()
+        f_out.close()
+
+        # write to user_nl_clm data if specified
+        if self.create_user_mods:
+            with open(os.path.join(user_mods_dir, "user_nl_clm"), "a") as nl_clm:
+                line = "landuse = '${}'".format(os.path.join(USRDAT_DIR, fluse_out))
+                self.write_to_file(line, nl_clm)
+
+    def create_surfdata_at_point(self, indir, file, user_mods_dir):
+        """
+        Create surface data file at a single point.
+        """
+        logger.info("----------------------------------------------------------------------")
+        logger.info(
+            "Creating surface dataset file at %s, %s", self.plon.__str__(), self.plat.__str__())
+
+        # specify file
+        fsurf_in = os.path.join(indir, file)
+        fsurf_out = add_tag_to_filename(fsurf_in, self.tag)
+        logger.info("fsurf_in:  %s", fsurf_in)
+        logger.info("fsurf_out: %s", os.path.join(self.out_dir, fsurf_out))
+
+        # create 1d coordinate variables to enable sel() method
+        f_in = self.create_1d_coord(fsurf_in, "LONGXY", "LATIXY", "lsmlon", "lsmlat")
+
+        # extract gridcell closest to plon/plat
+        f_out = f_in.sel(lsmlon=self.plon, lsmlat=self.plat, method="nearest")
+
+        # expand dimensions
+        f_out = f_out.expand_dims(["lsmlat", "lsmlon"]).copy(deep=True)
+
+
+        # modify surface data properties
+        if self.overwrite_single_pft:
+            f_out["PCT_NAT_PFT"][:, :, :] = 0
+            if self.dom_pft < 16:
+                f_out['PCT_NAT_PFT'][:, :, self.dom_pft] = 100
+        if not self.include_nonveg:
```

Comment:
> * Can users define an arbitrary PFT/PFT or PFT/CFT split for a grid cell?

Okay! I think this is a great idea to add the capability of mulitple PFTS for a grid cell. 
For example, something like this if we assume equal percentage for each pft.
```
if self.dom_pft is not None:
            f_out["PCT_NAT_PFT"][:, :, :] = 0
            percent_pft = 100/self.dom_pft
            for dpft in self.dom_pft:
                  f_out['PCT_NAT_PFT'][:, :, self.dom_pft] = percent_pft
```
The major point, will be how do you vision this implemented? I am thinking the command line argument for --dompft should have the capability to accept more than one dom-pft. 

For the percentage of pfts, (for example %60 and 40%) we need to introduce another flag to accept these values. For example: `--dompft 10 5 --pct_pft 40 60` or something like that.

Reply:
This makes sense to me. Is it also useful for FATES @jkshuman & @adrifoster?

Reply:
@negin513 and @wwieder I think this sounds useful and makes a lot of sense. I like the suggested implementation. To be clear these percentages refer to the amount of area within the natural veg cell allocated to each PFT? If that is the case, depending on the implementation in FATES (SP vs full competition) being able to assign these areas will be quite useful. For FATES-SP that area allocation will be important, and testing these different areas could be an interesting test case.

Reply:
Thanks @jkshuman. 

@negin513 let's let users define the pfts and their weights as you suggest above, realizing that this can be modified down the road as we build up functionality for FATES.

The example above makes sense for a mixed grid cell.
`--dompft 10,5` `--pct_pft 40,60`  

How do we handle a 'pure' grid cell
`--dompft 10` can pct_pft default to 100 if only one dompft is provided?

Reply:
Thanks so much for your comments @jkshuman and @wwieder.

Yes, that should be possible. Default would be 100%.
We should put in some error checking to make sure that : 
1. total of pct_pft is equal to 100.
2. The equal number of arguments are provided for `--dompft` and `--pct_pft` are equal. For example, something like `--dompft 10 --pct_pft 40,60` would not work.

Reply:
@wwieder to clarify my point, in most FATES runs users define a PFT file and the number of PFTs you want to use in competition, and FATES PFTs use 100% of the natural veg area. But with certain SP modes the PFT area is set by the surface dataset. For FATES-SP having the ability to intentionally modify these areas would be useful in the same way it is useful for Big Leaf runs.

Reply:
@ekluzek can you advise what else we need to do for simulations including crops

Reply:
This is documented in #1606. I think this section should have some testing around it, either system testing or unit-testing.

- [ ] Add testing for this tricky bit of code about crop and no crop.

Reply:
@negin513 and I talked about this and did some pair programming to come up with a possible solution for a unit test. @negin513 is working on isolating the tricky part of this into a method of the class that she can add unit testing for by setting up an x-array object that's sent in and then modified. The unit test can then ensure that the subroutine works correctly based on the subset data settings and the initial xarray object sent into it.  Having this unit test in place ensures that this is setup correctly and will continue to work correctly even as it changes.

I really like this ability to set an xarray object in a unit test and then test that it's being changed correctly by methods that operate on it. This prevents having to have files for the unit tests and all the problems that files bring, and will be useful for other unit tests as xarray is something that we utilize in the python code. So I'm looking forward to having this in place as it will expand our python unit testing capability. And provide some really useful examples.

I'm also going to add a crop test to the tools/test framework, so we'll also test both cases that way.